core/domain/topic_domain.py

```diff
@@ -827,6 +843,24 @@ def update_name(self, new_name):
         """
         self.name = new_name
 
+    def update_abbreviated_name(self, new_abbreviated_name):
+        """Updates the abbreviated_name of a topic object.
+
+        Args:
+            new_abbreviated_name: str. The updated abbreviated_name
+            for the topic.
+        """
+        self.abbreviated_name = new_abbreviated_name
+
+    def update_thumbnail_data_url(self, new_thumbnail_data_url):
+        """Updates the thumbnail_data_url of a topic object.
+
+        Args:
+            new_thumbnail_data_url: str. The updated thumbnail_data_url
```

Comment:
Could also be None right? (If the image gets removed.)

Reply:
Done.

Reply:
Done.